src/test-tracker.ts

```diff
@@ -14,34 +15,34 @@ export function startFile(file: string) {
 export function finishFile() {
   log(`---- END FILE ${currentFile}\n`);
   currentFile = '';
-  currentSampleId = '';
+  currentSampleId = undefined;
 }
 
-export function startSample(sampleId: string) {
+export function startSample(sampleId: IdMetadata) {
   currentSampleId = sampleId;
-  results[currentFile][currentSampleId] = 0;
+  results[currentFile][currentSampleId.key] = 0;
   sampleStartMs = Date.now();
   log(`BEGIN #${sampleId}\n`);
```

Comment:
Is this going to come out as `[Object object]`?

Reply:
Aha, good catch

Reply:
Aha, good catch! And a good case for https://typescript-eslint.io/rules/no-base-to-string...